src/main/java/com/scaleunlimited/flinkcrawler/pojos/CrawlStateUrl.java

```diff
@@ -47,16 +47,17 @@ public FetchStatus getStatus() {
     }
 
     public void setStatus(FetchStatus status) {
-        if (status != _previousStatus) {
```

Comment:
I'm not really sure what I was thinking here, but I remember worrying that I didn't necessarily want to update _previousStatus if setStatus was called a second time. Regardless, though, I would always want to ensure that _status ended up with status. Your code clearly fixes that bug (as well as the copy construction and equals bugs).